digitalaidseattle/seattle-humane-app
# Hotfix/0.2.2.18/refactor path aliases

## What type of PR is this? (check all applicable)


## Description

refactor any instances of relative paths that can instead be turned into path aliases through tsconfig. added a couple of new path aliases in the config as well. 

added an eslint plugin to have path aliases be a part of the linting process, and allows to auto convert them to path aliases using lint --fix

## Related Tickets & Documents

[user story 1](https://projects.digitalaidseattle.org/project-tasks-details?recordId=recOWcAvt4x22gfA8)
[user story 2](https://projects.digitalaidseattle.org/project-tasks-details?recordId=recnZsP22Ahm1dH03)

## QA Instructions, Screenshots, Recordings

The eslint plugin should catch any relative import paths and show an error.

```javascript
// example of code that will produce error:
import foo from "./greet"; // Should use "@/lib"
import bar from "../constants/hello.i18n.js"; // Sould use "@/constants"
```

```javascript
//example of code that won't produce errors
import foo from "@/lib/greet";
import bar from "@/constants/hello.i18n.js";
```

then you can use yarn lint --fix to automatically change those relative paths to aliases

find more info [here](https://github.com/msfragala/eslint-plugin-path-alias)

## Added/updated tests?
_We are aiming for 100% statement coverage._

- [x] No, and this is why: this is code refactoring related to import links and all changes are covered under linting rules 

## [optional] Which Seattle Humane animal would you like to dedicate this PR to?

![Stanley Yelnats ](https://g.petango.com/photos/337/e7cc8f69-25ec-4de8-bfdb-47f97d3f7688.jpg)
